pkg/controllers/monitoring-stack/components.go

```diff
@@ -288,6 +288,33 @@ func stackComponentPatchers(ms *stack.MonitoringStack, instanceSelectorKey strin
 				return desired, nil
 			},
 		},
+		{
+			empty: func() client.Object {
+				service := newThanosSidecarService(ms, instanceSelectorKey, instanceSelectorValue)
+				return &corev1.Service{
```

Comment:
while testing this, I found changes to services aren't reconciled, perhaps this patch can add a `Owns(&rbacv1.Role{}).WithEventFilter(p)` to the controller.

Reply:
:+1: good catch.